components/content-service/pkg/storage/minio.go

```diff
@@ -310,17 +312,21 @@ func (rs *DirectMinIOStorage) Upload(ctx context.Context, source string, name st
 	return
 }
 
-func minioBucketName(ownerID string) string {
+func minioBucketName(ownerID, bucketName string) string {
+	if bucketName != "" {
+		return bucketName
+	}
+
 	return fmt.Sprintf("gitpod-user-%s", ownerID)
 }
 
-func minioWorkspaceBackupObjectName(workspaceID string, name string) string {
-	return fmt.Sprintf("workspaces/%s/%s", workspaceID, name)
+func minioWorkspaceBackupObjectName(ownerID, workspaceID, name string) string {
+	return filepath.Join(ownerID, "workspaces", workspaceID, name)
```

Comment:
With this change, we change the name of the workspace bucket object name, correct? Wouldn't this introduce a backward-incompatible change? We would not be able to open workspace backups that has been created prior this change, right?

I think, we need to distinguish between the cases where a bucket name is given and where not.

Reply:
> We would not be able to open workspace backups that has been created prior this change, right?

Correct. There is no automatic migration or "hybrid mode" where we fall back to the old behavior.

Reply:
this should be fixed now.